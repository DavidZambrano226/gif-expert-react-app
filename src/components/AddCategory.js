import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {
    
    const [inputValue, setInputValue] = useState('');
    
    const handlerChange = (e) => {
        setInputValue(e.target.value)
    }

    const handlerSubmit = (e) => {
        e.preventDefault();
        
        if (inputValue.trim().length > 2 ) {
            setCategories( category => [inputValue, ...category] );
            setInputValue('');
        }

    }
    
    return (
        <form onSubmit={handlerSubmit}>
            <input 
                type="text" 
                value = {inputValue}
                onChange={ handlerChange } 
                
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
