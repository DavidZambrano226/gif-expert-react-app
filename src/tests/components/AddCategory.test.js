import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('AddCategory component tests', () => {

    const setCategories = jest.fn();
    let wrapper;

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    })

    test('should make match with the snapshot', () => {
        expect( wrapper ).toMatchSnapshot();
    })
    
    test('should change the value in the input text', () => {
        const input = wrapper.find('input');
        const value = 'Hello World';
        input.simulate('change', { target: { value } });        
    })
    
    test('should not call the setInput function', () => {
        const form = wrapper.find('form');

        form.simulate('submit', { preventDefault(){} });

        expect( setCategories ).not.toHaveBeenCalled();

    })

    test('should call to submit function in the form', () => {
        const value = 'Hello from submit';
        
        wrapper.find('input').simulate('change', { target: { value } });
        wrapper.find('form').simulate('submit', { preventDefault(){} });        
        

        expect( setCategories ).toHaveBeenCalled();
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) );

        expect( wrapper.find('input').prop('value') ).toBe('');

    })
    
    

})
