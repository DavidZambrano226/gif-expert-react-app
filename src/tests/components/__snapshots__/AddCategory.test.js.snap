// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory component tests should make match with the snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
